### PR TITLE
support compare x-ms-paths and paths 

### PR DESCRIPTION
**Pull request by jianyexi**

### CODE REVIEW

File: src/lib/validators/openApiDiff.ts [hunk 1]
```diff
@@ -23,6 +27,8 @@ export type Options = {
 export type ProcessedFile = {
   readonly fileName: string
   readonly map: sourceMap.BasicSourceMapConsumer | sourceMap.IndexedSourceMapConsumer
+  readonly resolvedFileName:string
```
Comment (ruowan) [inline src/lib/validators/openApiDiff.ts:30]: reformat the code.

File: src/lib/util/utils.ts [hunk 2]
```diff
@@ -541,3 +541,43 @@ export function relaxModelLikeEntities(model: Model) {
...
 }
+
+/**
+ * 
+ * @param jsonPath e.g paths['test!'].get.response
+ * @returns jonsPointer  /paths/test!/get/response
+ */
+
+export function pathToJsonPointer(jsonPath:string):string {
+  const replaceAllReg = (src:string):RegExp =>{
```
Comment (ruowan) [inline src/lib/util/utils.ts:552]: Where is the source of this code? It's hard to read. Pls put some doc or comments here.

File: src/lib/util/resolveSwagger.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  })
+  return target
+}
+
+export class ResolveSwagger {
+    innerSwagger:json.Json | undefined
+    file:string
+
+    constructor(file:string) {
+        this.file = file.replace('\\','/')
```
Comment (ruowan) [inline src/lib/util/resolveSwagger.ts:65]: Why replace here? Node standard path library supports some cross-platform function.

File: src/lib/util/resolveSwagger.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    stringify(): string {
+       return json.stringify(this.innerSwagger as json.JsonObject)
+    }
+
+    generateNew() {
+        writeFileSync(this.getResolvedPath(),this.stringify())
+    }
+
+    parse(url:string,data:string){
```
Comment (ruowan) [inline src/lib/util/resolveSwagger.ts:100]: code format.

File: src/lib/util/resolveSwagger.ts (hunk 3 above)
Comment (jianyexi) [reply src/lib/util/resolveSwagger.ts:65]: just  make our output path have the same format

Comment (ruowan) [reply src/lib/util/resolveSwagger.ts]: I mean you can use path.resolve to make file format in different platform. Windox and Linux have different paths format .

Comment (jianyexi) [reply src/lib/util/resolveSwagger.ts]: great . fix it

File: src/lib/util/utils.ts [hunk 5]
```diff
@@ -541,3 +541,45 @@ export function relaxModelLikeEntities(model: Model) {
   }
   return model
 }
+
+/**
+ * 
+ * @param jsonPath e.g paths['test!'].get.response
+ * @returns jonsPointer  /paths/test!/get/response
```
Comment (ruowan) [inline src/lib/util/utils.ts:548]: typo jons  => json. Does jsonPointer have a standard format? Any doc about jsonPointer

Comment (jianyexi) [reply src/lib/util/utils.ts]: fixded

**Pull request merged**